pkg/sloop/ingress/kubewatcher_test.go

```diff
@@ -81,21 +82,58 @@ func Test_bigPicture(t *testing.T) {
 	masterURL := "url"
 	kubeContext := "" // empty string makes things work
 	enableGranularMetrics := true
-	kw, err := NewKubeWatcherSource(kubeClient, outChan, resync, includeCrds, time.Duration(10*time.Second), masterURL, kubeContext, enableGranularMetrics)
+	exclusionRules := map[string][]any{
+		"_all": []any{
+			map[string]any{
+				"==": []any{
+					map[string]any{
+						"var": "metadata.name",
+					},
+					"s2",
+				},
+			},
+		},
+	}
+
+	kw, err := NewKubeWatcherSource(kubeClient, outChan, resync, includeCrds, time.Duration(10*time.Second), masterURL, kubeContext, enableGranularMetrics, exclusionRules)
 	assert.NoError(t, err)
 
-	// create service and await corresponding event
+	// create namespace
 	ns := "ns"
 	_, err = kubeClient.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}}, metav1.CreateOptions{})
 	if err != nil {
 		t.FailNow()
 	}
-	svc := &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "s"}}
+
+	// create first service
+	svc := &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "s1"}}
+	_, err = kubeClient.CoreV1().Services(ns).Create(context.TODO(), svc, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating service: %v\n", err)
+	}
+
+	// create second service, should be filtered out by exclusion rule
+	svc = &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "s2"}}
 	_, err = kubeClient.CoreV1().Services(ns).Create(context.TODO(), svc, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating service: %v\n", err)
 	}
-	_ = <-outChan
+
+	// create third service
+	svc = &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "s3"}}
+	_, err = kubeClient.CoreV1().Services(ns).Create(context.TODO(), svc, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating service: %v\n", err)
+	}
+
+	// await events
+	result1 := <-outChan
+	result2 := <-outChan
+	result3 := <-outChan
+
+	if !strings.Contains(result1.Payload, `"name":"ns"`) { t.Error("Expected event not found") }
+	if !strings.Contains(result2.Payload, `"name":"s1"`) { t.Error("Expected event not found") }
+	if !strings.Contains(result3.Payload, `"name":"s3"`) { t.Error("Expected event not found") }
```

Comment:
should we check that `"name":"s2"` is not present?

Reply:
Could you give an example? I found this the most straightforward way. I changed the `if` statements to `assert` calls, bit cleaner.